src/components/DocExplorer.js

```diff
@@ -148,6 +181,142 @@ export class DocExplorer extends React.Component {
   }
 }
 
+// Render Search Results
+class SearchDoc extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return (this.props.schema !== nextProps.schema) ||
+      (this.props.searchValue !== nextProps.searchValue);
+  }
+
+  render() {
+    const searchValue = this.props.searchValue;
+    const schema = this.props.schema;
+    const onClickType = this.props.onClickType;
+    const onClickField = this.props.onClickField;
+
+    const typeMap = schema.getTypeMap();
+
+    // Compose a search index for all possible arguments from every types.
+    let possibleArguments = [];
+    Object.keys(typeMap).forEach(typeName => {
+      let type = typeMap[typeName];
+      if (type.getFields) {
+        let fields = type.getFields();
+        Object.keys(fields).forEach(fieldName => {
+          let field = fields[fieldName];
+          if (field.args && field.args.length) {
+            Array.prototype.push.apply(possibleArguments, field.args);
+          }
+        });
+      }
+    });
+
+    let matchedTypes = [];
+    let matchedFields = [];
+    let matchedArgs = [];
+
+    Object.keys(typeMap).forEach(typeName => {
+      const type = typeMap[typeName];
+      let matchedOn = [];
+      if (typeName.indexOf(searchValue) !== -1) {
+        matchedOn.push('Type Name');
+      }
+
+      if (type.description && type.description.indexOf(searchValue) !== -1) {
+        matchedOn.push('Description');
+      }
+
+      if (matchedOn.length) {
+        matchedTypes.push(
+          <div className="doc-category-item">
+            <TypeLink type={type} onClick={onClickType} />
+          </div>
+        );
+      }
+
+      if (type.getFields) {
+        let fields = type.getFields();
+        Object.keys(fields).forEach(fieldName => {
+          let field = fields[fieldName];
+          if (fieldName.indexOf(searchValue) !== -1) {
+            matchedFields.push(
+              <div className="doc-category-item">
+                <a className="field-name"
+                  onClick={event => onClickField(field, type, event)}>
+                  {field.name}
+                </a>
+                {' on '}
+                <TypeLink type={type} onClick={onClickType} />
+              </div>
+            );
+          }
+
+          if (field.args && field.args.length) {
+            let matches = field.args.filter(arg => {
+              return arg.name.indexOf(searchValue) !== -1;
+            });
+            Array.prototype.push.apply(matchedArgs, matches.map(arg =>
```

Comment:
Since you're rendering fields here, you might want to add them to `matchedFields`. Also, rather than rendering the same field multiple times for each matched arg, you should just render all the matched args within this one field, moving the `matches.map` inside the `( )` selection below.